### PR TITLE
Add Ambire wallet

### LINKED ISSUE #2049: Add Ambire wallet
Add WalletConnect based Ambirre wallet using web3-react

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Fixes #2049

This adds a separate Ambire wallet option that uses a wallet-connect connector

Screenshot 2023-02-23 at 13 59 14

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @nenadV91 , great job! I have been able to run wrap/unwrap/approve TXs, place sell marken and But limit order. 

Some questions/issues: 
1. Just FYI: Ambire is a SC-wallet. So a user
 - can't place ETH orders
 - can't cancel orders
 - user can't place gasless orders (so would be great to show +gas in the Fees section) #2033 
 image

2. network selector is hidden when user is connected to this wallet (however, the app supports multiple networks for the same address. So it would great to add it
3. I don't think that the button makes sense in a mobile view: it is not user-friendly to press on the wallet option, see the list of wallets where there is no such wallet, then open another tab in the modal, copy the code, and only then open another browser's tab with the wallet account and paste there the WC link. See the video with this: 


https://user-images.githubusercontent.com/70885163/220952346-ea87b9f3-6a99-419c-b031-5f64b114bee0.mp4

@anxolin , WDYT about it? Maybe we should show Ambire wallet option only for desktop version? 

4.  it would be great to hide this option when the app is opened in MM integrated browser  (related to #1826 issue)

image

File: src/custom/hooks/useIsActiveWallet.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export const useIsActiveWallet = (connection: Connection) => {
   if (!isActive) {
     return false
   } else if (isActive && !selectedWallet) {
-    return true
+    return false
```
Comment (anxolin) [inline src/custom/hooks/useIsActiveWallet.ts:12]: why this change?

Comment (nenadV91) [reply src/custom/hooks/useIsActiveWallet.ts:12]: Because if I'm connected to WC wallet then I connect to Metamask, then I disconnect Metamask. It will show me that I'm connected to WC but I'm not really until I click on some WC wallet again.

Screenshot 2023-02-27 at 13 03 12

Comment (nenadV91) [reply src/custom/hooks/useIsActiveWallet.ts]: Actually will update this

**Comment by nenadV91** [comment]
@elena-zh 
1. Nothing to change for now as I understand
2. Even if we enable the selector it wont work properly, same as gnosis safe
3. Not sure about this one but we can decide later on and change this
4. I think we can leave this one for now

**Comment by elena-zh** [comment]
Hey @nenadV91 

>2.  Even if we enable the selector it wont work properly, same as gnosis safe

Then, I'd change the behavior of the selector: currently, when connect to the Ambire wallet, selector is displayed until a user places a transaction. Then, it becomes hidden. 
So my point is here: 
1. Whether we completely hide the selector when connected to Ambire wallet
2. OR we always show it (but you have mentioned that it is impossible). 
So, could you please remove it then at all? 
image

> 4. I think we can leave this one for now

I would agree, but currently, WC option is not displayed in the list of wallets in MM integrated browser. So I don't think it is a good idea to add other options into the list. So I'd remove them anyways. 

Thanks!

**Comment by nenadV91** [comment]
@elena-zh Good points, I think both 2 and 4 should be fixed now.

**Review by elena-zh — APPROVED** [review]
Hey @nenadV91 , thanks!
Approving given the fact that the 3rd case is still opened for discussion in a separate doc.

**Review by alfetopito — COMMENTED** [review]
Code looks mostly ok, although it seems there are unrelated changes.
My guess is the base branch was merged and you need to rebase against develop?

File: src/cow-react/common/hooks/useIsSmartContractWallet.ts [hunk 2]
```diff
@@ -30,10 +31,13 @@ export function useIsSmartContractWallet(): boolean {
 
   const isArgentWallet = useIsArgentWallet()
   const isSmartContract = useCheckIsSmartContract()
+  const isAmbireWallet = useIsAmbireWallet()
 
   useEffect(() => {
     if (!account) {
```
Comment (shoom3301) [inline src/cow-react/common/hooks/useIsSmartContractWallet.ts:37]:
```suggestion
    if (!account) {
      setIsSmartContractWallet(false)
      return
    }
    
    if (isAmbireWallet || isArgentWallet || isSmartContract) {
      setIsSmartContractWallet(true)
    }
```

File: src/cow-react/common/hooks/useIsSmartContractWallet.ts [hunk 3]
```diff
@@ -30,10 +31,13 @@ export function useIsSmartContractWallet(): boolean {
...
       setIsSmartContractWallet(false)
+    } else if (account && isAmbireWallet) {
+      setIsSmartContractWallet(true)
     } else if (account && isArgentWallet) {
       setIsSmartContractWallet(true)
     } else if (account && isSmartContract) {
```
Comment (shoom3301) [inline src/cow-react/common/hooks/useIsSmartContractWallet.ts:43]:
```suggestion
    if (!account) {
      setIsSmartContractWallet(false)
      return
    }
    
    if (isAmbireWallet || isArgentWallet || isSmartContract) {
      setIsSmartContractWallet(true)
    }
```

File: src/cow-react/common/hooks/useIsSmartContractWallet.ts (hunk 2 above)
Comment (shoom3301) [inline src/cow-react/common/hooks/useIsSmartContractWallet.ts:37]: Can be simplified:

```
    if (!account) {
      setIsSmartContractWallet(false)
      return
    }
    
    if (isAmbireWallet || isArgentWallet || isSmartContract) {
      setIsSmartContractWallet(true)
    }
```

File: src/cow-react/modules/wallet/api/components/WalletModal/options/AmbireOption.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { Connector } from '@web3-react/types'
+import AMBIRE_ICON_URL from 'assets/images/ambire.svg'
+import { ConnectionType, walletConnectConnection } from 'connection'
+import { getConnectionName } from 'connection/utils'
+import { useIsActiveWallet } from 'hooks/useIsActiveWallet'
+
+import Option from '../Option'
+import { useWalletInfo } from '../../../hooks/useWalletInfo'
```
Comment (shoom3301) [inline src/cow-react/modules/wallet/api/components/WalletModal/options/AmbireOption.tsx:8]: Imports shouldn't use relative paths

File: src/cow-react/modules/wallet/api/components/WalletModal/options/WalletConnectOption.tsx [hunk 5]
```diff
@@ -13,12 +16,18 @@ const BASE_PROPS = {
 }
 
 export function WalletConnectOption({ tryActivation }: { tryActivation: (connector: Connector) => void }) {
```
Comment (shoom3301) [inline src/cow-react/modules/wallet/api/components/WalletModal/options/WalletConnectOption.tsx:18]: Just thoughts, no need to address them now:
Looks like many `[WalletName]Option` components have the same boilerplate, can we get rid of the code duplications?
Smth like:
```
<WalletOption getWallet={getIsZengoWallet}></WalletOption>
<WalletOption getWallet={getIsAmbireWallet}></WalletOption>
...
```

File: src/custom/theme/baseTheme.tsx [hunk 6]
```diff
@@ -444,9 +444,9 @@ export const ThemedGlobalStyle = createGlobalStyle`
   }
 
   // START - Modal overrides
-  ${HeaderText} {
+  /* ${HeaderText} {
```
Comment (shoom3301) [inline src/custom/theme/baseTheme.tsx:447]: Do we need this in the future?
If yes, then would be great to add a comment, otherwise, just delete it

File: src/hooks/useIsAmbireWallet.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { useWalletMetaData } from '@cow/modules/wallet'
+import { getIsAmbireWallet } from '@src/custom/connection/utils'
```
Comment (shoom3301) [inline src/hooks/useIsAmbireWallet.ts:2]:
```suggestion
import { getIsAmbireWallet } from 'connection/utils'
```

File: src/hooks/useIsAmbireWallet.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+
+export default function useIsAmbireWallet(): boolean {
+  const { walletName } = useWalletMetaData()
+
+  return Boolean(getIsAmbireWallet(walletName))
```
Comment (shoom3301) [inline src/hooks/useIsAmbireWallet.ts:7]: Redundant cast to boolean. `getIsAmbireWallet` already has `boolean` return type

**Comment by nenadV91** [comment]
Closing in favor of new PR https://github.com/cowprotocol/cowswap/pull/2096
